modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/components/page-accessibility-tool/components/Violations.js

```diff
@@ -0,0 +1,234 @@
+	{label: Liferay.Language.get('critical'), value: 'critical'},
+	{label: Liferay.Language.get('serious'), value: 'serious'},
+	{label: Liferay.Language.get('moderate'), value: 'moderate'},
+	{label: Liferay.Language.get('minor'), value: 'minor'},
+];
+
+const CATEGORY_FILTER_OPTIONS = [
+	{label: 'WCAG 2.0A', value: 'wcag2a'},
+	{label: 'WCAG 2.0 Level AA', value: 'wcag2aa'},
+	{label: 'WCAG 2.1 Level A', value: 'wcag21a'},
+	{label: 'WCAG 2.1 Level AA', value: 'wcag21aa'},
+	{label: 'Best Practices', value: 'best-practice'},
+];
+
+function ViolationsFilter() {
+	const [dropdownExpanded, setDropdownExpanded] = useState(false);
+	const dispatch = useFilteredViolationsDispatch();
+	const {selectedCategories, selectedImpact} = useFilteredViolationsValues();
+
+	return (
+		<>
+			<ClayDropDown
+				active={dropdownExpanded}
+				closeOnClickOutside
+				onActiveChange={setDropdownExpanded}
+				trigger={
+					<ClayButtonWithIcon
+						displayType="unstyled"
+						small
+						symbol="filter"
+					/>
+				}
+			>
+				<ClayDropDown.ItemList>
+					<ClayDropDown.Group
+						header={Liferay.Language.get('filter-by-impact')}
+					>
+						{IMPACT_FILTER_OPTIONS.map((item) => {
+							const existsImpact =
+								selectedImpact.findIndex(
+									(element) => element === item.value
+								) !== -1;
+
+							return (
+								<ClayDropDown.Section
+									active={existsImpact}
+									key={item.value}
+								>
+									<ClayCheckbox
+										active={existsImpact}
+										label={item.label}
+										onChange={() => {
+											if (existsImpact) {
+												dispatch({
+													payload: item.value,
+													type: TYPES.IMPACT_REMOVE,
+												});
+
+												return;
+											}
+
+											dispatch({
+												payload: item.value,
+												type: TYPES.IMPACT_ADD,
+											});
+										}}
+									></ClayCheckbox>
+								</ClayDropDown.Section>
+							);
+						})}
+					</ClayDropDown.Group>
+					<ClayDropDown.Group
+						header={Liferay.Language.get('filter-by-category')}
+					>
+						{CATEGORY_FILTER_OPTIONS.map((item) => {
+							const existsCategory =
+								selectedCategories.findIndex(
+									(element) => element === item.value
+								) !== -1;
+
+							return (
+								<ClayDropDown.Section key={item.value}>
+									<ClayCheckbox
+										active={existsCategory}
+										label={item.label}
+										onChange={() => {
+											if (existsCategory) {
+												dispatch({
+													payload: item.value,
+													type: TYPES.CATEGORY_REMOVE,
+												});
+
+												return;
+											}
+											dispatch({
+												payload: item.value,
+												type: TYPES.CATEGORY_ADD,
+											});
+										}}
+									></ClayCheckbox>
+								</ClayDropDown.Section>
+							);
+						})}
+					</ClayDropDown.Group>
+				</ClayDropDown.ItemList>
+			</ClayDropDown>
+		</>
+	);
+}
+
+function ViolationsPanelHeaderTitle() {
```

Comment:
I think we can avoid creating another component, just to separate the markup, and move the markup where it is used.

For example, We may prefer creating another component:

- Reuse
- Shared for others components
- Single responsibility
- Separate logic

I think it is useful to make a decision on when to create a new component.